### DISCUSSION TITLE
The response type cannot be obtained using Hono RPC and TanStack Query

### DISCUSSION THREAD

**Discussion by igmtdev** [honojs/hono#3690] [Q&A]
I was trying to get the "return success" and "return error", but what I've got is this "type ResponseType = {}" or "type ResponseType = any"

![Screenshot from 2024-11-19 14-56-30](https://github.com/user-attachments/assets/1bcb1d35-5dd9-4224-bb95-427e530737f9)
![Screenshot from 2024-11-19 14-57-02](https://github.com/user-attachments/assets/c609c65e-54df-4e1c-bb25-820981dd74fe)

**Comment by vinayak-vohra** [comment]
@igmtdev I faced this same issue recently. Here is how you can solved it: 
```typescript
// type
type ResponseType = InferResponseType<(typeof client.api.auth)['update-profile']['$post'], 200>;
```
```typescript
// mutationFn
async ({ json }) => {
    const response = await client.api.auth['update-profile'].$post({ json });
    const resp = await response.json();

    if ('error' in resp) {
        throw new Error(resp.error);
    }

    return resp;
}
```

This is not a concrete solution, just a hack.
Also, I would encourage you to paste code instead of image : )
